spec/rubocop/cop/sorbet/obsolete_strict_memoization_spec.rb

```diff
@@ -0,0 +1,200 @@
+            @foo ||= some_really_long_initialization_expression______________________________________
+          end
+        RUBY
+
+        expect_correction(<<~RUBY)
+          sig { returns(SomeReallyLongTypeName______________________________________) }
+          def foo
+            @foo ||= T.let(
+              some_really_long_initialization_expression______________________________________,
+              T.nilable(SomeReallyLongTypeName______________________________________),
+            )
+          end
+        RUBY
+
+        autocorrected_source = autocorrect_source(source)
+        longest_line = autocorrected_source.lines.max_by(&:length)
+        expect(longest_line.length).to be <= 120
+      end
+    end
+
+    describe "with multiline initialization expression" do
+      it "registers an offence and autocorrects into a multiline expression" do
+        # There's special auto-correct logic to handle a multiline initialization expression, so that it
+        # *doesn't* end up like this:
+        #
+        #   sig { returns(Foo) }
+        #   def foo
+        #     @foo = T.let(multiline_method_call(
+        #       foo,
+        #       bar,
+        #       baz,
+        #     ), T.nilable(Foo))
+        #   end
+
+        source = <<~RUBY
+          sig { returns(Foo) }
+          def foo
+            @foo = T.let(@foo, T.nilable(Foo))
+            @foo ||= multiline_method_call(
+              foo,
+              bar,
+              baz,
+            )
+          end
+        RUBY
+
+        corrected_source = <<~RUBY
+          sig { returns(Foo) }
+          def foo
+            @foo ||= T.let(
+              multiline_method_call(
+                foo,
+                bar,
+                baz,
+              ),
+              T.nilable(Foo),
+            )
+          end
+        RUBY
+
+        expect(autocorrect_source(source)).to(eq(corrected_source))
+      end
+
+      describe "with a gap between the two lines" do
+        it "registers an offence and autocorrects into a multiline expression" do
+          source = <<~RUBY
+            sig { returns(Foo) }
+            def foo
+              @foo = T.let(@foo, T.nilable(Foo))
+      
+              @foo ||= multiline_method_call(
+                foo,
+                bar,
+                baz,
+              )
+            end
+          RUBY
+
+          corrected_source = <<~RUBY
+            sig { returns(Foo) }
+            def foo
+              @foo ||= T.let(
+                multiline_method_call(
+                  foo,
+                  bar,
+                  baz,
+                ),
+                T.nilable(Foo),
+              )
+            end
+          RUBY
+
+          expect(autocorrect_source(source)).to(eq(corrected_source))
+        end
+      end
+    end
+
+    context "when its not the first line in a method", pending: "Not implemented yet" do
```

Comment:
I'm not sure how to support this. I can start my node matcher with `...` to metch against `some`/`other`/`code`, but then the replacement will completely delete those lines.

If I do match against them, I'd need to tweak the replacement to skip over them and only replace the correct portion.

More generally, what's the right way to match "any two consecutive lines, anywhere" (ignoring whitespace in between)?

Reply:
That's an interesting idea!

I'll ask on the Rubocop project, and see if there's a "blessed" way to express this, otherwise that seems like a promising fallback!